.github/workflows/test-lint.yaml

```diff
@@ -28,3 +28,6 @@ jobs:
 
       - name: Check shell scripts
         run: make shellcheck
+
+      - name: Run pre-commit
+        uses: pre-commit/action@v3.0.1
```

Comment:
It looks like that recommended way is to use pre-commit CI: https://github.com/pre-commit/action?tab=readme-ov-file#pre-commitaction.
Are we planning to migrate to this CI ?

Reply:
I figured it would be faster to execute here since checkout and python setup are already being executed [here](https://github.com/kubeflow/katib/blob/fa0552ac7f598cd5f4d1a85378a8068e3622ec68/.github/workflows/test-lint.yaml#L17-L24), but open to moving it into a discrete workflow.

Reply:
In case of GitHub actions execution how these pre-commits are added to the PRs ?
For instance, when we use the pre-commit CI, it adds commit with `pre-commit` author.

Reply:
My understanding is that this action simply runs pre-commit and, if there are any findings or modified files, the CI gate won't pass and the PR will not be mergeable, i.e. this action won't actually modify the code / PR. It's a way to ensure that contributors are running pre-commit locally before submitting PRs, optimizing code standards and consistency in a way that should make life easier for PR reviewers.

Reply:
I purposely didn't run the hooks after rebasing, so some imports are out of order, and here's the resulting CI gate failing because of a finding in pre-commit (for reference, I'll fix this momentarily):
https://github.com/kubeflow/katib/actions/runs/9940660170/job/27457887876?pr=2242

Reply:
That makes, sense. Thanks for explaining.